lib/feature/localization/bloc/localization_bloc.dart

```diff
@@ -0,0 +1,52 @@
+import 'dart:async';
+
+import 'package:app/app/service/localization/localization.dart';
+import 'package:bloc/bloc.dart';
+import 'package:freezed_annotation/freezed_annotation.dart';
+import 'package:logging/logging.dart';
+
+part 'localization_event.dart';
+part 'localization_state.dart';
+part 'localization_bloc.freezed.dart';
+
+class LocalizationBloc extends Bloc<LocalizationEvent, LocalizationState> {
+  LocalizationBloc(this.localizationService)
+      : super(
+          LocalizationState(
+            localeCode: localizationService.localeCode,
+          ),
+        ) {
+    on<LocalizationEvent>((event, emit) {
```

Comment:
Imho instead of `event.when` it would be easier to read, if you use `on< _SetLocaleCodeFromService >` and `on< _ChangeLocaleCode >`

Reply:
agree, 🤟